app/models/manageiq/providers/azure/cloud_manager/event_parser.rb

```diff
@@ -11,10 +11,11 @@ def self.event_to_hash(event, ems_id)
     event_hash = {
       :source     => "AZURE",
       :timestamp  => event["eventTimestamp"],
-      :message    => event["description"].presence,
+      :message    => event["description"].presence || event["operationName"]["localizedValue"].presence,
       :ems_id     => ems_id,
       :event_type => event_type,
-      :full_data  => event
+      :full_data  => event,
+      :ems_ref    => event["eventDataId"]
```

Comment:
It might be the matter of taste, but I'm always trying to have a comma on the such last line (ruby allows that), because it will be looking nicer on the diff next time somebody else will add some new line. You can see as example here line above marked as changed because of that comma.

Reply:
I think it's a habit from the past. If I remember correctly, older versions of Ruby would choke on trailing commas in lists.

Reply:
> because it will be looking nicer on the diff next time somebody else will add some new line

^ I do the same thing, rubocop started complaining about it recently but I hate seeing the previous line in a diff when you just add something after it.

Not sure if this is one of those new rubocop rules that just got added automatically or if it was a conscious decision

Reply:
If the rule changes, or we want to override it locally, we can update it. But for now I'm just gonna go with Rubocop's recommendations.